roles/kube-node-common/files/auto-join-node.sh

```diff
@@ -2,7 +2,27 @@
 
 source /cloud-init-report-alert.sh
 
-
+function err_retry() {
+  local exit_code=$1
+  local attempts=$2
+  local sleep_millis=$3
+  shift 3
+  for attempt in `seq 1 $attempts`; do
+    if [[ $attempt -gt 1 ]]; then
+      echo "Attempt $attempt of $attempts"
+    fi
+    # This weird construction lets us capture return codes under -o errexit
+    "$@" && local rc=$? || local rc=$?
+    if [[ ! $rc -eq $exit_code ]]; then
+      return $rc
+    fi
+    if [[ $attempt -eq $attempts ]]; then
+      return $rc
+    fi
+    local sleep_ms="$(($attempt * $attempt * $sleep_millis))"
+    sleep "${sleep_ms:0:-3}.${sleep_ms: -3}"
```

Comment:
Can't we just divide by 1000? Do we really need sub-second precision?

Reply:
+1